docs/sources/reference/components/otelcol/otelcol.receiver.solace.md

```diff
@@ -0,0 +1,210 @@
+---
+canonical: https://grafana.com/docs/alloy/latest/reference/components/otelcol/otelcol.receiver.solace/
+description: Learn about otelcol.receiver.solace
+title: otelcol.receiver.solace
+---
+
+# otelcol.receiver.solace
+
+`otelcol.receiver.solace` accepts traces from a [Solace PubSub+ Event Broker](https://solace.com/products/event-broker/) and
+forwards it to other `otelcol.*` components.
+
+> **NOTE**: `otelcol.receiver.solace` is a wrapper over the upstream
+> OpenTelemetry Collector `solace` receiver from the `otelcol-contrib`
+> distribution. Bug reports or feature requests will be redirected to the
+> upstream repository, if necessary.
+
+Multiple `otelcol.receiver.solace` components can be specified by giving them
+different labels.
+
+## Usage
+
+```alloy
+otelcol.receiver.solace "LABEL" {
+  queue = "QUEUE"
+  auth {
+    // sasl_plain or sasl_xauth2 or sasl_external block
+  }
+  output {
+    traces  = [...]
+  }
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+| Name                 | Type     | Description                                                               | Default          | Required |
+| -------------------- | -------- | ------------------------------------------------------------------------- | ---------------- | -------- |
+| `queue`              | `string` | Name of the Solace telemetry queue to get span trace messages from.       |                  | yes      |
+| `broker`             | `string` | Name of the Solace broker using amqp over tls.                            | `localhost:5671` | no       |
```

Comment:
Should we consider keeping this the same as upstream, so when upstream is changed to utilize multiple broker support we don't need to change the config in Alloy?

Reply:
To support multiple brokers, they would need to change the structure of the config because with several brokers you probably want the possibility to specify a queue and credentials for each broker. Also they should change the name to "brokers" for consistency and change the doc since they refer to it as "the broker" + the component is already "old" and considered "stable" in otel standard. So I don't see this change happening in the near future / happening without other breaking changes in the config.
I would prefer to ask upstream if they are ok to switch to a string and if not, whether they are really planning to support multiple brokers at some point

Reply:
Upstream contribution: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36387